src/Microsoft.TestPlatform.Extensions.BlameDataCollector/BlameCollector.cs

```diff
@@ -215,63 +219,67 @@ private void CollectDumpAndAbortTesthost()
+            }
+
+            if (_uploadDumpFiles)
+            {
+                try
                 {
-                    try
+                    var dumpFiles = _processDumpUtility.GetDumpFiles(true, /* if we killed it by hang dumper, we already have our dump, otherwise it might have crashed, and we want all dumps */ !hangDumpSuccess);
+                    foreach (var dumpFile in dumpFiles)
                     {
-                        if (!string.IsNullOrEmpty(dumpFile))
+                        try
                         {
-                            var fileTransferInformation = new FileTransferInformation(_context.SessionDataCollectionContext, dumpFile, true, _fileHelper);
-                            _dataCollectionSink.SendFileAsync(fileTransferInformation);
+                            if (!string.IsNullOrEmpty(dumpFile))
+                            {
+                                var fileTransferInformation = new FileTransferInformation(_context.SessionDataCollectionContext, dumpFile, true, _fileHelper);
+                                _dataCollectionSink.SendFileAsync(fileTransferInformation);
+                            }
+                        }
+                        catch (Exception ex)
+                        {
+                            // Eat up any exception here and log it but proceed with killing the test host process.
+                            EqtTrace.Error(ex);
                         }
-                    }
-                    catch (Exception ex)
-                    {
-                        // Eat up any exception here and log it but proceed with killing the test host process.
-                        EqtTrace.Error(ex);
-                    }
 
-                    if (!dumpFiles.Any())
-                    {
-                        EqtTrace.Error("BlameCollector.CollectDumpAndAbortTesthost: blame:CollectDumpOnHang was enabled but dump file was not generated.");
+                        if (!dumpFiles.Any())
+                        {
+                            EqtTrace.Error("BlameCollector.CollectDumpAndAbortTesthost: blame:CollectDumpOnHang was enabled but dump file was not generated.");
+                        }
                     }
                 }
+                catch (Exception ex)
+                {
+                    _logger.LogError(_context.SessionDataCollectionContext, $"Blame: Collecting hang dump failed with error.", ex);
+                }
             }
-            catch (Exception ex)
+            else
             {
-                _logger.LogError(_context.SessionDataCollectionContext, $"Blame: Collecting hang dump failed with error.", ex);
+                EqtTrace.Info("BlameCollector.CollectDumpAndAbortTesthost: Custom path to dump directory was provided via VSTEST_DUMP_PATH. Skipping attachment upload, the caller is responsible for collecting and uploading the dumps themselves.");
```

Comment:
[minor][pedantic] Not entirely sure here but shouldn't it be `itself/himself/herself` instead of `themselves` as we have one caller?

Reply:
It could be, but because we are not sure about the gender and we don't need to imply one, we can use a gender-neutral pronoun "they" which is valid in singular form: https://en.wikipedia.org/wiki/Singular_they

Reply:
Oh thanks didn't know about this.